install/kubernetes/cilium/templates/cilium-operator/deployment.yaml

```diff
@@ -137,11 +137,10 @@ spec:
         - name: AZURE_CLIENT_SECRET
           value: {{ .Values.azure.clientSecret }}
         {{- end }}
+        {{- end }}      
+        {{- with .Values.operator.extraEnv }}
+        {{- toYaml . | nindent 8 }}
         {{- end }}
-        {{- range $key, $value := .Values.operator.extraEnv }}
-        - name: {{ $key }}
-          value: {{ $value }}
-        {{- end }}        
```

Comment:
Changing this from an object to a list is backward compatible?

Reply:
Yes. But wait, if you look at `extraEnv` in everywhere else, there is mismatch between `values.yaml` and implementations in templates. It always define as object but implement as list.

https://github.com/cilium/cilium/blob/c40ed791109d6b1500b2a86c2a75126e507a934e/install/kubernetes/cilium/values.yaml#L124

https://github.com/cilium/cilium/blob/c40ed791109d6b1500b2a86c2a75126e507a934e/install/kubernetes/cilium/templates/cilium-agent/daemonset.yaml#L202-L204